src/GovUk.Education.ExploreEducationStatistics.Data.Model.Tests/Repository/FootnoteRepositoryTests.cs

```diff
@@ -13,247 +15,105 @@ namespace GovUk.Education.ExploreEducationStatistics.Data.Model.Tests.Repository
-                }
-            };
+            var release = _releaseGenerator.Generate();
+            
+            var subject1Filters = _filterGenerator
+                .WithFilterGroups(_ => _filterGroupGenerator
+                    .WithFilterItems(_ => _filterItemGenerator.GenerateList(1))
+                    .GenerateList(1))
+                .GenerateList(3);
+            
+            var (subject1, subject2) = _subjectsGenerator
+                .ForIndex(0, s => s.SetFilters(subject1Filters))
+                .GenerateTuple2();
+
+            var subject1IndicatorGroup1 = _indicatorGroupGenerator
+                .WithIndicators(_indicatorGenerator.GenerateList(1))
+                .WithSubject(subject1)
+                .Generate();
+
+            var footnotes = _footnoteGenerator
+                .ForIndex(0, s => s
+                    .Set(f => f.Content, "Applies to subject 1")
+                    .SetSubjects(_ => ListOf(subject1)))
+                .ForIndex(1, s => s
+                    .Set(f => f.Content, "Applies to subject 1 filter 1")
+                    .SetFilters(_ => ListOf(subject1Filters[0])))
+                .ForIndex(2, s => s
+                    .Set(f => f.Content, "Applies to subject 1 filter 2 group 1")
+                    .SetFilterGroups(_ => ListOf(subject1Filters[1].FilterGroups[0])))
+                .ForIndex(3, s => s
+                    .Set(f => f.Content, "Applies to subject 1 filter 3 group 1 item 1")
+                    .SetFilterItems(_ => ListOf(subject1Filters[2].FilterGroups[0].FilterItems[0])))
+                .ForIndex(4, s => s
+                    .Set(f => f.Content, "Applies to subject 1 indicator 1")
+                    .SetIndicators(_ => ListOf(subject1IndicatorGroup1.Indicators[0])))
+                
+                // Footnote applies to subject 1 filter 1
+                // and subject 1 filter 2 group 1
+                // and subject 1 filter 3 group 1 item 1
+                // and subject 1 indicator 1
+                .ForIndex(5, s => s
+                    .Set(f => f.Content, "Applies to multiple attributes of subject 1")
+                    .SetFilters(_ => ListOf(subject1Filters[0]))
+                    .SetFilterGroups(_ => ListOf(subject1Filters[1].FilterGroups[0]))
+                    .SetFilterItems(_ => ListOf(subject1Filters[2].FilterGroups[0].FilterItems[0]))
+                    .SetIndicators(_ => ListOf(subject1IndicatorGroup1.Indicators[0])))
+                .ForIndex(6, s => s
+                    .Set(f => f.Content, "Applies to subject 2")
+                    .SetSubjects(_ => ListOf(subject2)))
+                .GenerateList();
+            
+            var releaseFootnotes = _releaseFootnoteGenerator
+                .ForInstance(s => s.Set(rf => rf.Release, release))
+                .ForInstance(s => s.Set(rf => rf.Footnote, (_, _, context) => footnotes[context.Index]))
+                .GenerateList(footnotes.Count);
```

Comment:
Similar to `FootnoteServicesTests`, with some additional extensions, we could roll this all up into something like:

```cs
var releaseFootnotes = _releaseFootnoteGenerator
   .WithRelease(release)
   .ForIndex(0, s => s
       .SetFootnote(_fixture.DefaultFootnote()
           .Set(f => f.Content, "Applies to subject 1"))
           .SetSubjects(_ => ListOf(subject1)))
   .ForIndex(1, s => s
       .SetFootnote(_fixture.DefaultFootnote()
           .Set(f => f.Content, "Applies to subject 1 filter 1"))
           .SetFilters(_ => ListOf(subject1Filters[0])))
   // etc...
   .GenerateList(5);
```

This negates the need to associate each footnote using `context.Index`.

Similar thing applies for other test cases.

Reply:
Yup happy to do this

Reply:
I'd personally say that to keep getting the benefits that the defaults give us (in terms of Footnotes, things like an incrementing Order value and Content string), we'd be best to go with the `ReleaseFootnoteGeneratorExtensions.WithFootnotes(List<Footnote>)` method, that would allow us to provide a List of Footnotes and have them applied to each ReleaseFootnote generated based upon their index in the List.  Then we can either create the Footnotes separately as before, or in situ within this WithFootnotes method, and still ahve the benefits of generating them as a single sequence.

Reply:
All done for ReleaseFootnotes and ReleaseSubjects (and in turn, incorporating creation of Filters and IndicatorGroups inline as well).  See example:

```
            var releaseSubject = _fixture
                .DefaultReleaseSubject()
                .WithRelease(release)
                .WithSubject(_fixture
                    .DefaultSubject()
                    .WithIndicatorGroups(_fixture
                        .DefaultIndicatorGroup()
                        .WithIndicators(_fixture.DefaultIndicator().Generate(3))
                        .GenerateList(1))
                    .Generate())
                .Generate();

            var indicatorGroup = releaseSubject.Subject.IndicatorGroups[0];
            
            var releaseFootnotes = _fixture.DefaultReleaseFootnote()
                .WithRelease(release)
                .WithFootnotes(_fixture
                    .DefaultFootnote()
                    .ForIndex(0, s => s
                        .SetContent("Applies to indicator 1")
                        .SetIndicators(ListOf(indicatorGroup.Indicators[0])))
                    .ForIndex(1, s => s
                        .SetContent("Applies to indicator 3")
                        .SetIndicators(ListOf(indicatorGroup.Indicators[2])))
                    .ForIndex(2, s => s
                        .SetContent("Applies to indicator 1 and indicator 3")
                        .SetIndicators(ListOf(indicatorGroup.Indicators[0], indicatorGroup.Indicators[2])))
                    .GenerateList())
                .GenerateList();
```